### PR TITLE
Neonichu add rubocop to plugin template

### PR DESCRIPTION
**Pull request by KrauseFx**
- Adds symlink to shared `.rubocop.yml`
- Add `rake` and `rubocop` as development dependencies
- Add `rspec` and `rubocop` rake tasks
- Fix rubocop issues in the template

cc @neonichu 

Taken from https://github.com/fastlane/fastlane/pull/4951

### CODE REVIEW

File: fastlane/lib/fastlane/plugins/plugin_generator.rb [hunk 1]
```diff
@@ -82,5 +84,12 @@ def replace_path_variables(template_path, plugin_info)
 
       path
     end
+
+    # This method takes care of any cleaning up we have to do
+    # after copying over all the files
+    def after_all(dest_path)
+      # Prefix the rubocop.yml file with a dot, since it's a rubocop requirement
+      FileUtils.mv(File.join(dest_path, "rubocop.yml"), File.join(dest_path, ".rubocop.yml"))
+    end
```
Comment (mfurtak) [inline fastlane/lib/fastlane/plugins/plugin_generator.rb]: This doesn't feel like it should be necessary... We already copy a file from the template whose name starts with a dot. (.rspec) Let's chat about this

File: fastlane/fastlane.gemspec [hunk 2]
```diff
@@ -3,6 +3,12 @@ lib = File.expand_path('../lib', __FILE__)
 $LOAD_PATH.unshift(lib) unless $LOAD_PATH.include?(lib)
 require 'fastlane/version'
 
+# Copy over the latest .rubocop.yml style guide
+# This will use the file name without the `.` as 
+# hidden files are ignored
+rubocop_config = File.expand_path('../../.rubocop.yml', __FILE__)
+`cp #{rubocop_config} #{lib}/fastlane/plugins/template/rubocop.yml`
```
Comment (mfurtak) [inline fastlane/fastlane.gemspec]: Evaluating this as part of fastlane's gemspec feels weird to me. Can we talk about why this is necessary?

File: fastlane/lib/fastlane/plugins/plugin_generator.rb (hunk 1 above)
Comment (neonichu) [inline fastlane/lib/fastlane/plugins/plugin_generator.rb]: Unfortunately, those files are ignored when building the gem, though, copying them over only works while developing at the moment

Comment (KrauseFx) [inline fastlane/lib/fastlane/plugins/plugin_generator.rb]: Just pushed a new commit together with @mfurtak to simplify this 👍

File: fastlane/fastlane.gemspec [hunk 3]
```diff
@@ -3,6 +3,12 @@ lib = File.expand_path('../lib', __FILE__)
 $LOAD_PATH.unshift(lib) unless $LOAD_PATH.include?(lib)
 require 'fastlane/version'
 
+# Copy over the latest .rubocop.yml style guide
+# This will use the file name without the `.` as 
+# hidden files are ignored
```
Comment (mfurtak) [inline fastlane/fastlane.gemspec]: The comment is now out of date 😊

**Comment by neonichu** [comment]
This one can be closed now as the work was merged in #4951

**Comment by KrauseFx** [comment]
Stealing my commits, as always

**Pull request closed without merging**